### ISSUE TITLE
[Bug]:  Geolocator has conflict with Safe Device package

### ISSUE THREAD

**Issue by rodrigo-altbank** [Baseflow/flutter-geolocator#1456]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [X] iOS
- [ ] Linux
- [ ] macOS
- [ ] Web
- [ ] Windows

### Steps to reproduce

1 - Addition of the Safe Device package;
2 - Addition of the Geolocator package;
3 - Accept the Share Location While In Use option, or give location permission to the App;

### Expected results

Application flows normally, without bottlenecks or delays

### Actual results

The application freezes for approximately 10 seconds, on some devices reaching 45 seconds, more evident on Android devices.

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or video

<details>
<summary>Screenshots or video demonstration</summary>

[Upload media here]

</details>

### Version

11.0.0

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Last login: Sun Mar 10 15:53:45 on ttys017
❯ flutter doctor -v
[!] Flutter (Channel [user-branch], 3.19.2, on macOS 14.4 23E214 darwin-arm64, locale en-BR)
    ! Flutter version 3.19.2 on channel [user-branch] at /Users/648/Programs/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at
      https://flutter.dev/docs/get-started/install.
    ! Upstream repository unknown source is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to unknown source to dismiss this error.
    • Framework revision 7482962148 (2 weeks ago), 2024-02-27 16:51:22 -0500
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git"
      directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/648/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C65
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.87.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • Mi 9 SE (mobile) • 9bc7b340                  • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • 00008110-000058313680A01E • ios            • iOS 17.3.1 21D61
    • macOS (desktop)  • macos                     • darwin-arm64   • macOS 14.4 23E214 darwin-arm64
    • Chrome (web)     • chrome                    • web-javascript • Google Chrome 122.0.6261.112

[✓] Network resources
    • All expected network resources are available.
```

</details>

**Comment by satella12** [comment]
how fixit ?

**Comment by TimHoogstrate** [comment]
Dear @rodrigo-altbank,

Did you reproduce it on iOS devices?

Kind regards,

**Comment by mvanbeusekom** [comment]
Hi @rodrigo-altbank,

This seemed to be an issue with the safe_device plugin which according to the author has been resolved with version 1.1.5 of the safe_device plugin (see this [issue](https://github.com/ufukhawk/safe_device/issues/42)).

I will go ahead and close this issue for now.
